ci/release/README.md

```diff
@@ -0,0 +1,140 @@
+- There is needed a `manual release` to send the libraries to `maven central`.
+- In case of issues/problems/errors there is a `Drop` options available to delete the release version.
+- Once a component has been released and published to the Central Repository, it cannot be altered.
+
+![release](img/release_process.png)
+
+## Artifacts
+
+- Github Artifacts:
+  - https://github.com/substrait/sustrait-java/releases
+
+- Sonatype OSSRH Artifacts:
+  - https://s01.oss.sonatype.org/content/repositories/{snapshots,staging,releases}/substrait/substrait-java/core/1.0.0/
+
+## Branches Configuration
+
+- In case of new features or breaking changes create a branch with any format name.
+  Example:
+  ```shell
+  # Current version on production: 2.5.7
+  $ git checkout -b new_feature_1234 main
+  ```
+  After your PR is merged to main branch, you could see the new libraries available at Sonatype staging environment.
+
+
+- In case of maintenances please create a branch name with this pattern `+([0-9])?(.{+([0-9]),x}).x`.
+  Example:
+  ```shell
+  # Current version on production: 2.5.7, version planned to use for maintenance: 2.1.7
+  $ git checkout -b 2.1.x v2.1.7 # for fix
+  $ git checkout -b 2.x v2.1.7 # for maintenance
+  ```
+  After your PR is merged to main branch, you could see the new libraries available at Sonatype staging environment.
+
+
+- In case you need to experiment with breaking change functionalities for new feature, take a look at pre-release branch.
+  Example:
+  ```shell
+  # Current version on production: 2.5.7, version planned to use for maintenance: 2.1.7
+  $ git checkout -b new_feature_lots_changes beta
+  ```
+  After your PR is merged to main branch, you could see the new libraries available at Sonatype snapshots environment.
+
+
+Branches configuration:
+
+```json
+  "branches": [
+    // maintenances (also generate release)
+    { "name": "+([0-9])?(.{+([0-9]),x}).x" },
+    // release
+    { "name": "main" },
+    { "name": "next" },
+    { "name": "next-major" },
+    // pre-release
+    { "name": "beta", "prerelease": true },
+    { "name": "alpha", "prerelease": true }
+  ],
+```
+
+## Q&A
+
+#### 1. What will be happened if I merge something on `main` branch and finalize with errors?
+
+CICD has a token configure with granted access to create issues as needed:
+
+```shell
+[10:30:55 PM] [semantic-release] [@semantic-release/github] › ℹ  Created issue #3: https://github.com/substrait/sustrait-java/issues/3.
+```
+
+![release_failing](img/automate_release_is_failing.png)
+
+#### 2. Could be possible to implement a process to deliver a library with custom version (i.e.: 3.2.9.RC1, 5.0.0.M1)?
+
+We are using Sonatype OSSRH and there are repositories for `snapshots` / `staging` / `release`:
+- If your library finalize with `1.4.3-XXXXX-SNAPSHOT` -> it is going to be deployed to Snapshots repository
+- If your library finalize with `1.4.3-XXXXX-SNAPSHOT.YYYY` -> it is going to be deployed to Staging/Releases repository
```

Comment:
What does it mean for a library to finalize?  Does this mean the destination repository is based on the name of the artifacts?  Or is the the value of the version field?

Reply:
Let me update that. The destination repository is based on the name of the artifacts. It will be:

We are using Sonatype OSSRH, the destination repository (`snapshots` / `staging` / `release`) is based on the name of the artifacts:
- For library name artifacts as `1.4.3-XXXXX-SNAPSHOT` -> it is going to be deployed to Snapshots repository
- For library name artifacts as `1.4.3-XXXXX-SNAPSHOT.YYYY` -> it is going to be deployed to Staging/Releases repository
- For library name artifacts as `1.4.3-XXXXX` -> it is going to be deployed to Staging/Releases repository
- For library name artifacts as `1.4.3` -> it is going to be deployed to Staging/Releases repository